slick/src/main/scala/slick/jdbc/PositionedResult.scala

```diff
@@ -32,6 +32,7 @@ abstract class PositionedResult(val rs: ResultSet) extends Closeable { outer =>
   final def <<? [T](implicit f: GetResult[Option[T]]): Option[T] = if(hasMoreColumns) this.<< else None
 
   final def nextBoolean()    = { val npos = pos + 1; val r = rs getBoolean    npos; pos = npos; r }
+  final def nextChar()       = { val npos = pos + 1; val r = rs getString     npos; pos = npos; if(r eq null) null else r.charAt(0) }
```

Comment:
Also, I saw this fails to compile anyway since null or Char are now of type Any (which is why I  rebased the commit away, thus the outdated diff) But you are right, it would discard those characters. My thought was that Char wouldn't have been the column's type if the datatype of the column was Char(1) or Varchar(1), but it might need that assert anyway.